crates/ruff_python_parser/src/token.rs

```diff
@@ -536,45 +539,28 @@ impl TokenKind {
         matches!(self, TokenKind::Newline | TokenKind::NonLogicalNewline)
     }
 
+    /// Returns `true` if the token is a keyword (including soft keywords).
+    ///
+    /// See also [`TokenKind::is_soft_keyword`], [`TokenKind::is_non_soft_keyword`].
     #[inline]
-    pub const fn is_keyword(self) -> bool {
-        matches!(
-            self,
-            TokenKind::False
-                | TokenKind::True
-                | TokenKind::None
-                | TokenKind::And
-                | TokenKind::As
-                | TokenKind::Assert
-                | TokenKind::Await
-                | TokenKind::Break
-                | TokenKind::Class
-                | TokenKind::Continue
-                | TokenKind::Def
-                | TokenKind::Del
-                | TokenKind::Elif
-                | TokenKind::Else
-                | TokenKind::Except
-                | TokenKind::Finally
-                | TokenKind::For
-                | TokenKind::From
-                | TokenKind::Global
-                | TokenKind::If
-                | TokenKind::Import
-                | TokenKind::In
-                | TokenKind::Is
-                | TokenKind::Lambda
-                | TokenKind::Nonlocal
-                | TokenKind::Not
-                | TokenKind::Or
-                | TokenKind::Pass
-                | TokenKind::Raise
-                | TokenKind::Return
-                | TokenKind::Try
-                | TokenKind::While
-                | TokenKind::With
-                | TokenKind::Yield
-        )
+    pub fn is_keyword(self) -> bool {
+        TokenKind::And <= self && self <= TokenKind::Type
```

Comment:
I worry that this is somewhat fragile — if we added `TokenKind::Aardvark` in the future because of some new Python syntax, we might forget to update this method, which could possibly cause subtle bugs. But I think my comment here also applies to the current implementation of this method, so I think this is fine

Reply:
Yeah, I think it's fine unless Python introduces a very obscure keyword ;)

(This change was part of the previous PR #11445)